core/src/Kokkos_View.hpp

```diff
@@ -1708,6 +1708,36 @@ class View : public ViewTraits<DataType, Properties...> {
   }
 };
 
+// These deduction guides result in an unmanaged View
+template <typename T>
+View(T*)->View<T>;
+
+template <typename T>
+View(T*, size_t)->View<T*>;
+
+template <typename T>
+View(T*, size_t, size_t)->View<T**>;
+
+template <typename T>
+View(T*, size_t, size_t, size_t)->View<T***>;
+
+template <typename T>
+View(T*, size_t, size_t, size_t, size_t)->View<T****>;
+
+template <typename T>
+View(T*, size_t, size_t, size_t, size_t, size_t)->View<T*****>;
+
+template <typename T>
+View(T*, size_t, size_t, size_t, size_t, size_t, size_t)->View<T******>;
+
+template <typename T>
+View(T*, size_t, size_t, size_t, size_t, size_t, size_t, size_t)
+    ->View<T*******>;
+
+template <typename T>
+View(T*, size_t, size_t, size_t, size_t, size_t, size_t, size_t, size_t)
+    ->View<T********>;
```

Comment:
General questions after thinking about these: are they too dangerous? Should we make it use AnonymousSpace instead? I.e. its type erased memory space as opposed to just assuming you gave me Default memory space?